src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -7089,5 +7089,143 @@ class C
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void ForEachComponentStatement()
+        {
+            string src1 = @"
+foreach(var (x1, x2) in new[] { (1, 2) }) { Console.Write(1); }
+";
+
+            string src2 = @"
+foreach(var (x3, x4) in new[] { (1, 2) }) { Console.Write(1); }
+foreach(var (x1, x2) in new[] { (30, 40) }) { Console.Write(50); }
+";
+
+            // variable names weight more in matching
+            var expected = new MatchingPairs
+            {
+                { "foreach(var (x1, x2) in new[] { (1, 2) }) { Console.Write(1); }", "foreach(var (x1, x2) in new[] { (30, 40) }) { Console.Write(50); }" },
+                { "{ Console.Write(1); }", "{ Console.Write(50); }" },
+                { "Console.Write(1);", "Console.Write(1);" }
+            };
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void ForWithDeconstruction()
+        {
+            string src1 = @"
+for (var (x1, x2) = (1, 2); ; ) { Console.Write(1); }
+";
+
+            string src2 = @"
+for (var (x3, x4) = (1, 2); ; ) { Console.Write(1); }
+for (var (x1, x2) = (30, 40); ; ) { Console.Write(50); }
+";
+
+            // variable names weight more in matching
+            var expected = new MatchingPairs
+            {
+                { "for (var (x1, x2) = (1, 2); ; ) { Console.Write(1); }", "for (var (x1, x2) = (30, 40); ; ) { Console.Write(50); }" },
+                { "var (x1, x2) = (1, 2)", "var (x1, x2) = (30, 40)" },
+                { "{ Console.Write(1); }", "{ Console.Write(50); }" },
+                { "Console.Write(1);", "Console.Write(1);" }
+            };
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void For()
+        {
+            string src1 = @"
+for (var x1 = 1; ; ) { Console.Write(1); }
+";
+
+            string src2 = @"
+for (var x3 = 1; ; ) { Console.Write(1); }
+for (var x1 = 30; ; ) { Console.Write(50); }
+";
+
+            // variable names weight more in matching
+            var expected = new MatchingPairs
+            {
+                { "for (var x1 = 1; ; ) { Console.Write(1); }", "for (var x1 = 30; ; ) { Console.Write(50); }" },
+                { "var x1 = 1", "var x1 = 30" },
+                { "x1 = 1", "x1 = 30" },
+                { "{ Console.Write(1); }", "{ Console.Write(50); }" },
+                { "Console.Write(1);", "Console.Write(1);" }
+            };
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        [WorkItem(13904, "https://github.com/dotnet/roslyn/issues/13904")]
+        public void Assignment()
```

Comment:
You're adding a test ensuring we get the wrong behavior. Can you please place the correct behavior in the test and mark it skipped with this URI?